File: frontend/bukasol/app/Models/MuhasabahReport.php
<?php

namespace App\Models;

use App\Models\Student;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class MuhasabahReport extends Model
{
    use HasFactory;

    protected $table = 'muhasabah_reports';
    protected $primaryKey = 'id';
    public $incrementing = true;
    protected $keyType = 'integer';
    public $timestamps = true;

    protected $fillable = [ 
        'student_id',
        'time_stamp',
        'surah_name',
        'surah_ayat',
        'sunnah_pray',
        'subuh_pray',
        'dzuhur_pray',
        'ashar_pray',
        'maghrib_pray',
        'isya_pray',
        'teacher_sign',
        'parent_sign',
    ];

    protected function casts () : array
    {
        return [ 
            'id'           => 'integer',
            'student_id'   => 'integer',
            'time_stamp'   => 'date',
            'surah_name'   => 'string',
            'surah_ayat'   => 'string',
            'sunnah_pray'  => 'boolean',
            'subuh_pray'   => 'boolean',
            'dzuhur_pray'  => 'boolean',
            'ashar_pray'   => 'boolean',
            'maghrib_pray' => 'boolean',
            'isya_pray'    => 'boolean',
            'teacher_sign' => 'boolean',
            'parent_sign'  => 'boolean',
        ];
    }

    public function student ()
    {
        return $this->belongsTo ( Student::class);
    }
}

File: server/bukasol/src/main/java/com/abdimas/bukasol/data/model/MuhasabahReport.java
package com.abdimas.bukasol.data.model;

import java.time.LocalDate;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "muhasabah_report")
public class MuhasabahReport {

    @Id
    @Column(name = "ID", columnDefinition = "BINARY(16)", updatable = false, nullable = false)
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private LocalDate timeStamp;

    @Column(nullable = true)
    private String surahName;

    @Column(nullable = true)
    private String surahAyat;

    @Column(nullable = false)
    private boolean sunnahPray;

    @Column(nullable = false)
    private boolean subuhPray;

    @Column(nullable = false)
    private boolean dzuhurPray;

    @Column(nullable = false)
    private boolean asharPray;

    @Column(nullable = false)
    private boolean maghribPray;

    @Column(nullable = false)
    private boolean isyaPray;

    @Column(nullable = false)
    private boolean teacherSign;

    @Column(nullable = false)
    private boolean parentSign;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;
}
